plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,313 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import io.airlift.log.Logger;
+import io.trino.Session;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.MoreFiles.deleteDirectoryContents;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.deltalake.DeltaLakeConnectorFactory.CONNECTOR_NAME;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogJsonEntryPath;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseDeltaLakeRegisterTableProcedureTest
+        extends AbstractTestQueryFramework
+{
+    private static final Logger LOG = Logger.get(BaseDeltaLakeRegisterTableProcedureTest.class);
+
+    protected static final String CATALOG_NAME = "test_delta";
+    protected static final String SCHEMA = "register_table_" + randomTableSuffix();
+
+    protected File metastoreDir;
+    protected HiveMetastore metastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(SCHEMA)
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+
+        this.metastoreDir = new File(queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake_data").toString());
+        this.metastore = createTestMetastore();
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin());
+
+        queryRunner.createCatalog(CATALOG_NAME, CONNECTOR_NAME, getConnectionProperties());
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+
+        return queryRunner;
+    }
+
+    protected abstract HiveMetastore createTestMetastore();
+
+    protected abstract Map<String, String> getConnectionProperties();
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        try {
+            if (metastore != null) {
+                // Data is on the local disk and will be deleted by the deleteOnExit hook
+                metastore.dropDatabase(SCHEMA, false);
+                deleteRecursively(metastoreDir.toPath(), ALLOW_INSECURE);
```

Comment:
redundant? (per above)

Reply:
Removed the comment. `deleteRecursively` is needed.